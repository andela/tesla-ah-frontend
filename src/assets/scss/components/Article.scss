@import '../abstracts/variables';

.article-card {
  padding: 15px 10px;
  background-color: #fff;
  margin-bottom: 10px;
  border-radius: 10px;
  box-shadow: 0 10px 6px -6px #777;
}

.article-card .d-avatar,
.follow-card .d-avatar {
  height: 50px;
  width: 50px;
}

.article-card-img {
  border-radius: 5px;
  height: 300px;
}

.article-title {
  color: #000;
  font-size: 2em;
}

.article-description,
.article-actions {
  color: #000;
  font-size: 1.3em;
}

a:hover {
  text-decoration: none;
}

.article-title:hover {
  text-decoration: underline;
}