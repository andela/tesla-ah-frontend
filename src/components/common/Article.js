import React from 'react';
import { Link } from 'react-router-dom';
import { trim } from 'lodash';

import '../../assets/scss/components/Article.scss';

const Article = (props) => {
  const {
    coverImage,
    title,
    description,
    lastUpdated,
    readTime,
    likes,
    dislikes,
    slug,
  } = props;
  return (
    <React.Fragment>
      <Link to={`/articles/${slug}`} className="article-card d-flex">
        <div className="flex-grow-1 d-flex flex-column">
          {coverImage ? (
            <img
              className="article-card-img img-fluid"
              src={trim(coverImage, "'")}
              alt="avatar"
            />
          ) : null}
          <p className="article-title align-self-start">{title}</p>
          <p className="article-description align-self-start">{description}</p>
          <div className="article-actions d-flex flex-column flex-sm-row justify-content-between">
            <span>{new Date(lastUpdated).toDateString()}</span>
            <span>
              {readTime}
              {' '}
              read
            </span>
            <span>
              <i className="fas fa-thumbs-up" />
              {' '}
              {likes}
            </span>
            <span>
              <i className="fas fa-thumbs-down" />
              {' '}
              {dislikes}
            </span>
          </div>
        </div>
      </Link>
    </React.Fragment>
  );
};

export default Article;
