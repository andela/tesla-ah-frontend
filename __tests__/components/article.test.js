/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { articles } from '../mockData';
import Article from '../../src/components/common/Article';

const renderArticleForm = (args) => {
  const defaultProps = {
    coverImage: '',
    title: '',
    description: '',
    lastUpdated: '',
    readTime: '',
    likes: '',
    dislikes: '',
    slug: '',
  };
  const props = { ...defaultProps, ...args };

  return shallow(<Article {...props} />);
};

it('renders article component', () => {
  const {
    gallery,
    title,
    description,
    updatedAt,
    readtime,
    slug,
  } = articles[0];
  const wrapper = renderArticleForm({
    coverImage: gallery[0],
    title,
    description,
    lastUpdated: new Date(updatedAt).toDateString(),
    readTime: readtime,
    likes: 10,
    dislikes: 5,
    slug,
  });

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.article-card').length).toBe(1);
  expect(wrapper.find('img').length).toBe(1);
  expect(wrapper.find('.article-title').length).toBe(1);
  expect(wrapper.find('.article-description').length).toBe(1);
  expect(wrapper.find('.article-actions').length).toBe(1);
  expect(wrapper.find('span').length).toBe(4);
});

it('renders article component without image', () => {
  const {
    title,
    description,
    updatedAt,
    readtime,
    slug,
  } = articles[0];

  const wrapper = renderArticleForm({
    coverImage: null,
    title,
    description,
    lastUpdated: new Date(updatedAt).toDateString(),
    readTime: readtime,
    likes: 10,
    dislikes: 5,
    slug,
  });

  expect(wrapper.find('img').length).toBe(0);
});
