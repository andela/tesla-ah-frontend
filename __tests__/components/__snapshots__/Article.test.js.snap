// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders article component 1`] = `ShallowWrapper {}`;
